_posts/2016/2016-02-09-microservice-meetup.md

```diff
@@ -0,0 +1,45 @@
+
+Eureka itself made a good impression but we discarded it along the way - also because back then it required code changes on the client side and we wanted something less intrusive.
+
+We tried to keep service discovery out of our code as much as possible and we added Docker to our infrastructure tool set. So the service discovery solution should seamlessly integrate into this environment as well. This lead us to a [Consul](https://www.consul.io/)-based service discovery approach that we already talked about in detail [earlier on this blog](https://developer.epages.com/blog/2015/09/28/service-discovery.html).
+
+Later we chose [Kubernetes](http://kubernetes.io/) as our container management solution and in Kubernetes we already found a DNS-based service discovery solution that could also fulfill our needs. So it felt strange to have our own service discovery stack running while Kubernetes offers something similar. That lead us to the adoption of Kubernetes also for service discovery. This helped us to reduce our infrastructure complexity and relieved us from managing additional components.
+
+On the service client side we introduced [Ribbon](https://github.com/Netflix/ribbon/wiki) for outbound service calls. Thus we can work with logical service names in a managed environment like Kubernetes but can also wire services together that run locally in a dev environment. Using the Spring framework the usage of Ribbon can be reduced to a few annotations enabling us to keep our service code (almost) free of service discovery concerns.
+
+{% image blog-microservice-meetup-3.jpg %}{% endimage %}
+
+After Dirk finished to guide us through this journey towards our current service discovery solution we took a break to recover with beer and pizza. There were excited discussions about the topic showing that service discovery is a topic that is of great interest in the community at the moment.
+
+{% image blog-microservice-meetup-2.jpg %}{% endimage %}
+
+After the pizza break Oli continued with a demo including live coding of two services demonstrating our service discovery solution in action. Oli not only showed how easy calling a service is on the development side, but also how scaling and health-checking can work in Kubernetes. You can find the [demo code on github](https://github.com/otrosien/meetup-2016-02-code).
+
+The slides of our talk can be found [here](http://epages-de.github.io/meetup-2016-02-slides/).
+
+We really like the idea of Meetups and are happy that we could participate. So it is not unlikely that more Meetups @epages will follow. Last but not least we would like to thank Thomas Hackbarth for organizing the Microservices Meetup Hamburg and for given us the oppportunity to host this event.
```

Comment:
... like to thank Thomas Hackbarth for organising the ... for giving us the opportunity...